Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -2173,7 +2173,12 @@ void WebPageProxy::updateActivityState(OptionSet<ActivityState::Flag> flagsToUpd
         m_activityState.add(ActivityState::IsVisibleOrOccluded);
     if (flagsToUpdate & ActivityState::IsInWindow && pageClient().isViewInWindow())
         m_activityState.add(ActivityState::IsInWindow);
-    if (flagsToUpdate & ActivityState::IsVisuallyIdle && pageClient().isVisuallyIdle())
+    bool isVisuallyIdle = pageClient().isVisuallyIdle();
+#if PLATFORM(COCOA) && !HAVE(CGS_FIX_FOR_RADAR_97530095) && ENABLE(MEDIA_USAGE)
+    if (pageClient().isViewVisible() && m_mediaUsageManager && m_mediaUsageManager->isPlayingVideoInViewport())
+        isVisuallyIdle = false;
```

Comment:
VisuallyIdle is used in a couple of other places:

* it causes us to drop the WebPageProxy::m_pageIsUserObservableCount token, in WebPageProxy::updateThrottleState
* it influences the rendering update frequency, in Page::setIsVisuallyIdleInternal

I think it's cleaner to make VisuallyIdle correct in the face of this issue with videos rather than add a new ActivityState::IsPlayingVideoInViewport flag and to check it in these places too.

Reply:
Ok, I'm not objecting to the approach. Would be good if this was reviewed by a media engineer though.